internal/serverinstall/k8s.go

```diff
@@ -140,11 +138,7 @@ func (i *K8sInstaller) Install(
 			return nil, err
 		}
 
-		var secretData struct {
-			Metadata struct {
-				Name string `yaml:"name"`
-			} `yaml:"metadata"`
-		}
+		var secretData apiv1.Secret
 
 		err = yaml.Unmarshal(data, &secretData)
```

Comment:
FYI, I'm not sure what's up with this but it isn't working exactly right. I'm testing with this config:

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: mysecret
type: Opaque
stringData:
  username: hello
  password: password
```

Which works with the previous changes, but with this it looks like it's unmarshaling to nothing and `secretData` ends up empty. :thinking:

Reply:
Looks like the potential issue here is we have to use a k8s client-go specific yaml parser to do this. I'll come back to this tomorrow.

Reply:
Actually I have an update for this. will resolve.